bindings/go/oci/internal/singlelayer/pack.go

```diff
@@ -0,0 +1,67 @@
+package singlelayer
```

Comment:
Why do we need this functionality in its own package?

Reply:
the packing of oci artifacts is quite complex for OCM due to the various ways we encode content. I thought an internal package makes this easier to encapsulate and test.